index.js

```diff
@@ -1,21 +1,542 @@
+      await exec.exec(plutilExportCommand, null, null);
 
-    core.debug((new Date()).toTimeString()); // debug is only output if you set the secret `ACTIONS_RUNNER_DEBUG` to true
-    await wait(parseInt(ms));
-    core.info((new Date()).toTimeString());
+      let jsonString = fs.readFileSync(testrunJson, "utf8");
+      const testTargets = JSON.parse(jsonString);
+
+      for (const target of Object.keys(testTargets)) {
+        if (target.charAt(0) !== "_") {
+          if (testTargets[target].TestingEnvironmentVariables) {
+            await insertEnvVariables(testRun, target);
+          } else if (target === "TestConfigurations") {
+            let configurationNumber = 0;
+            for (const configuration of testTargets["TestConfigurations"]) {
+              let testNumber = 0;
+              for (const test of configuration["TestTargets"]) {
+                await insertEnvVariables(
+                  testRun,
+                  target +
+                    "." +
+                    configurationNumber +
+                    "." +
+                    "TestTargets" +
+                    "." +
+                    testNumber
+                );
+              }
+            }
+          }
+        }
+      }
+      //run tests
+      let testCommand =
+        "xcodebuild test-without-building " +
+        codeCoverParam +
+        " -xctestrun " +
+        `"${testRun}"` +
+        ' -destination "' +
+        destination +
+        '"' +
+        extraParameters;
+      try {
+        await exec.exec(testCommand, null, null);
+      } catch (error) {
+        testError = error.message;
+      }
+    }
+    if (testError) {
+      core.setFailed(testError.message);
+      console.log(testError.message);
+    }
 
-    core.setOutput('time', new Date().toTimeString());
   } catch (error) {
     core.setFailed(error.message);
+    console.log(error.message);
+
+  }
+
+    //Clean up
+  fs.rmSync(sdkTestingDir, { recursive: true });
+
+}
+
+function getSDKForPlatform(platform) {
+  switch (platform) {
+    case "macos":
+    case "mac":
+      return "macosx";
+    case "tvos":
+      return "appletvsimulator";
+    default:
+      return "iphonesimulator";
+  }
+}
+
+function getDestinationForPlatform(platform) {
+  switch (platform) {
+    case "macos":
+    case "mac":
+      return "platform=macOS,arch=x86_64";
+    case "tvos":
+      return "platform=tvOS Simulator,name=Apple TV 4K";
+    default:
+      return "platform=iOS Simulator,name=iPhone 13";
   }
 }
 
+function getFrameworkPathForPlatform(platform) {
+  switch (platform) {
+    case "macos":
+    case "mac":
+      return "macos-arm64_x86_64 ";
+    case "tvos":
+      return "tvos-arm64_x86_64-simulator";
+    default:
+      return "ios-arm64_x86_64-simulator";
+  }
+}
+
+async function deleteLinesContaining(file, match) {
+  let newName = file + "_old";
+  await io.mv(file, newName);
+  fs.readFileSync(newName, { encoding: "utf-8" }, function(err, data) {
```

Comment:
I don't think this works. `readFileSync` does not have a callback because it's sync. It'd be something like: 
```javascript
const data = fs.readFileSync(newName, { encoding: "utf-8" })
```
and then continue with what you're doing in the `function(err, data)` callback

Reply:
ok, changed